thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/dataframe/util/DataFrameUtils.java

```diff
@@ -493,14 +518,14 @@ public long apply(long... values) {
    * Helper: Returns a pre-populated ThirdeyeRequestBuilder instance. Removes invalid filter values.
    *
    * @param slice metric data slice
-   * @param metric metric dto
    * @param dataset dataset dto
    * @param expressions metric expressions
-   * @param metricDAO metric config DAO
    * @return ThirdeyeRequestBuilder
    * @throws ExecutionException
    */
-  private static ThirdEyeRequest.ThirdEyeRequestBuilder makeThirdEyeRequestBuilder(MetricSlice slice, MetricConfigDTO metric, DatasetConfigDTO dataset, List<MetricExpression> expressions, MetricConfigManager metricDAO) throws ExecutionException {
+  private static ThirdEyeRequest.ThirdEyeRequestBuilder makeThirdEyeRequestBuilder(MetricSlice slice, DatasetConfigDTO dataset, List<MetricExpression> expressions) throws ExecutionException {
+    validateTimeSeriesSize(dataset, slice.getStart(), slice.getEnd(), slice.getGranularity());
```

Comment:
1. I think we can have this validation earlier when parsing YAML. 
2. Suggest to split the method into two - estimateTimeSeriesSize and validateTimeSeriesSize.
3. Can we add a limit when retrieving data? Like using LIMIT 10000. I remember we have this for Pinot.

Reply:
@akshayrai  Have you get a chance to address the comments?